docs/source/primitives.rst

```diff
@@ -156,27 +154,27 @@ Schema
 Update
 ******
 
-Update tasks are used to change values in the context. These tasks have three major forms:
+Update tasks are used to change values in the context. An update task itself may contain multiple distinct updates to perform, these essentially act as subtasks. The subtasks come in three major forms, dictated by the keywords used in their invocation as shown below:
 
-1. updating the context with a raw value:
+1. updating the context with a raw value, key word - "result":
 
 .. code-block::
 
     {"result": "x", "result_key": "$.destination"}
 
-2. renaming a context value:
+2. renaming a context value, key word - "key":
 
 .. code-block::
 
     {"key": "x", "result_key": "$.destination"}
 
-3. creating a new value using a string template:
+3. creating a new value using a string template - key word - "template":
 
 .. code-block::
 
     {"template": "{$.value}{$.another}", "result_key": "$.destination"}
 
-There are also extra flags ``append`` and ``extend``, which are mutually exclusive, they act by updating the destination (assumed to be a list) ASK_MICKY (are they always a list? This makes it sound too optional, we should state about cases where they aren't if they aren't).
+There are also extra flags ``append`` and ``extend``, which are mutually exclusive, they act by modifying the above major cases ASK_MICKY (is this correct?), updating a destination which must be a list.
```

Comment:
remove the ask

Reply:
extend requires the result you are trying to add on to also be a list, the difference between an append and an extend